### ISSUE TITLE
QSIPREP not picking up dwi images for recon workflows

### ISSUE THREAD

**Issue by julfou81** [PennLINC/qsiprep#662]
Hi,

Thank you for providing this nice tool!
I got many successful executions of qsiprep v0.18.1 for preprocessing and reconstruction for datasets with full PEPOLAR acquistions (same directions acquired in AP and PA).

For comparison purposes, I generated a dataset where only one phase encoding direction had all the diffusion directions measurements while I kept only two b0s in the other phase encoding direction.

For this new dataset, the preprocessing with qsiprep worked well, but when I launched the reconstruction part, I got the following error:

```
IMPORTANT:nipype.workflow:
    Running qsirecon version 0.18.1:
      * BIDS dataset path: /work/GesteSigne.
      * Participant list: ['test'].
      * Run identifier: 20231121-174300_28ec39ca-dfb4-4933-9edc-961d86d4fb08.
    
INFO:nipype.workflow:Found usable QSIPrep-preprocessed T1w image and mask.
INFO:nipype.workflow:Found T1w-to-template transforms
INFO:nipype.workflow:Freesurfer directory /work/GesteSigne/derivatives/fmriprep/sourcedata/freesurfer/sub-test exists for test
INFO:nipype.workflow:Anatomical (T1w) available for recon: {'has_qsiprep_5tt_hsvs': False, 'has_freesurfer_5tt_hsvs': False, 'has_freesurfer': True, 'has_qsiprep_t1w': True, 'has_qsiprep_t1w_transforms': True}
Process Process-2:
Traceback (most recent call last):
  File "/usr/local/miniconda/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/local/miniconda/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/miniconda/lib/python3.8/site-packages/qsiprep/cli/run.py", line 1180, in build_recon_workflow
    retval['workflow'] = init_qsirecon_wf(
  File "/usr/local/miniconda/lib/python3.8/site-packages/qsiprep/workflows/recon/base.py", line 89, in init_qsirecon_wf
    single_subject_wf = init_single_subject_wf(
  File "/usr/local/miniconda/lib/python3.8/site-packages/qsiprep/workflows/recon/base.py", line 232, in init_single_subject_wf
    dwi_recon_wfs[dwi_file] = init_dwi_recon_workflow(
  File "/usr/local/miniconda/lib/python3.8/site-packages/qsiprep/workflows/recon/build_workflow.py", line 45, in init_dwi_recon_workflow
    QsiReconIngress(dwi_file=dwi_file),
  File "/usr/local/miniconda/lib/python3.8/site-packages/nipype/interfaces/base/core.py", line 563, in __init__
    super(SimpleInterface, self).__init__(
  File "/usr/local/miniconda/lib/python3.8/site-packages/nipype/interfaces/base/core.py", line 202, in __init__
    self.inputs.trait_set(**inputs)
  File "/usr/local/miniconda/lib/python3.8/site-packages/traits/has_traits.py", line 1520, in trait_set
    setattr(self, name, value)
  File "/usr/local/miniconda/lib/python3.8/site-packages/nipype/interfaces/base/traits_extension.py", line 330, in validate
    value = super(File, self).validate(objekt, name, value, return_pathlike=True)
  File "/usr/local/miniconda/lib/python3.8/site-packages/nipype/interfaces/base/traits_extension.py", line 135, in validate
    self.error(objekt, name, str(value))
  File "/usr/local/miniconda/lib/python3.8/site-packages/traits/base_trait_handler.py", line 74, in error
    raise TraitError(
traits.trait_errors.TraitError: The 'dwi_file' trait of a QsiReconIngressInputSpec instance must be a pathlike object or string representing an existing file, but a value of '/made/up/outputs/sub-X_dwi.nii.gz' <class 'str'> was specified.

```

Here is the command I used for preprocessing, which executed with no error.
```
singularity run --cleanenv -B /scratch/jsein/BIDS:/work \
 --nv /scratch/jsein/my_images/qsiprep-0.18.1.sif /work/$study  \
 /work/$study/derivatives participant --participant_label $sub     \
 -w /work/temp_data_${study}  --output-resolution 1.2  --fs-license-file /work/freesurfer/license.txt \
  --eddy-config /work/$study/code/qsiprep/eddy_params.json   \
 --b0-threshold 50 --unringing-method mrdegibbs  --denoise-method dwidenoise  \
 --anat-modality T1w --distortion-group-merge none 
 ```

And the command I used for reconstruction that threw the error reported above:
```
singularity run  -B /scratch/jsein/BIDS:/work,$HOME/.templateflow:/home/qsiprep/.cache/templateflow \
--nv /scratch/jsein/my_images/qsiprep-0.19.0.sif /work/$study  \
 /work/$study/derivatives participant --participant_label $sub     \
 -w /work/temp_data_${study}  --fs-license-file /work/freesurfer/license.txt \
 --recon-only --recon-spec mrtrix_multishell_msmt_ACT-hsvs  --recon-input /work/$study/derivatives/qsiprep \
 --freesurfer-input /work/$study/derivatives/fmriprep/sourcedata/freesurfer  
```

Here the subject tree (where the `dir-PA_dwi.nii.gz` file contains only 2 b0s)
```
sub-test
├── anat
│   ├── sub-test_T1w.json
│   └── sub-test_T1w.nii.gz
└── dwi
    ├── sub-test_dir-AP_dwi.bval
    ├── sub-test_dir-AP_dwi.bvec
    ├── sub-test_dir-AP_dwi.json
    ├── sub-test_dir-AP_dwi.nii.gz
    ├── sub-test_dir-PA_dwi.bval
    ├── sub-test_dir-PA_dwi.bvec
    ├── sub-test_dir-PA_dwi.json
    └── sub-test_dir-PA_dwi.nii.gz
```
And here is the content of the qsiprep derivatives folder after preprocessing:

```
derivatives/qsiprep
├── dataset_description.json
├── dwiqc.json
├── logs
│   ├── CITATION.html
│   ├── CITATION.md
│   └── CITATION.tex
├── sub-test
│   ├── anat
│   │   ├── sub-test_desc-aseg_dseg.nii.gz
│   │   ├── sub-test_desc-brain_mask.nii.gz
│   │   ├── sub-test_desc-preproc_T1w.nii.gz
│   │   ├── sub-test_dseg.nii.gz
│   │   ├── sub-test_from-MNI152NLin2009cAsym_to-T1w_mode-image_xfm.h5
│   │   ├── sub-test_from-T1wACPC_to-T1wNative_mode-image_xfm.mat
│   │   ├── sub-test_from-T1wNative_to-T1wACPC_mode-image_xfm.mat
│   │   ├── sub-test_from-T1w_to-MNI152NLin2009cAsym_mode-image_xfm.h5
│   │   └── sub-test_from-orig_to-T1w_mode-image_xfm.txt
│   ├── dwi
│   │   ├── sub-test_confounds.tsv
│   │   ├── sub-test_desc-ImageQC_dwi.csv
│   │   ├── sub-test_desc-SliceQC_dwi.json
│   │   ├── sub-test_dwiqc.json
│   │   ├── sub-test_space-T1w_desc-brain_mask.nii.gz
│   │   ├── sub-test_space-T1w_desc-eddy_cnr.nii.gz
│   │   ├── sub-test_space-T1w_desc-preproc_dwi.b
│   │   ├── sub-test_space-T1w_desc-preproc_dwi.bval
│   │   ├── sub-test_space-T1w_desc-preproc_dwi.bvec
│   │   ├── sub-test_space-T1w_desc-preproc_dwi.nii.gz
│   │   ├── sub-test_space-T1w_desc-preproc_dwi.txt
│   │   └── sub-test_space-T1w_dwiref.nii.gz
│   ├── figures
│   │   ├── sub-test_carpetplot.svg
│   │   ├── sub-test_coreg.svg
│   │   ├── sub-test_desc-resampled_b0ref.svg
│   │   ├── sub-test_desc-sdc_b0.svg
│   │   ├── sub-test_dir-AP_dwi_denoise_dir_AP_dwi_wf_denoising.svg
│   │   ├── sub-test_dir-AP_dwi_denoise_dir_AP_dwi_wf_unringing.svg
│   │   ├── sub-test_dir-PA_dwi_denoise_dir_PA_dwi_wf_denoising.svg
│   │   ├── sub-test_dir-PA_dwi_denoise_dir_PA_dwi_wf_unringing.svg
│   │   ├── sub-test_final_denoise_wf_biascorr.svg
│   │   ├── sub-test_sampling_scheme.gif
│   │   ├── sub-test_seg_brainmask.svg
│   │   └── sub-test_t1_2_mni.svg
```

Thank you for your help!

**Comment by julfou81** [comment]
It turns out that I was making a mistake for this test. I did create an artificially new subject by copying the data of an existing subject and renaming the copy with a new subject name. This worked for preprocessing with qsiprep but not for qsirecon for some reason that I didn't catch yet.
Now I did save the original subject into another folder and made my test on the subject's modified data by keeping the same subject name and now qsiprep and qsirecon went through.
